source-boilerplate/boilerplate.go

```diff
@@ -384,7 +384,52 @@ func hwHashPartition(partitionId []byte) uint32 {
 	return uint32(highwayhash.Sum64(partitionId, highwayHashKey) >> 32)
 }
 
-func IncludesHwHash(range_ *pf.RangeSpec, partitionID []byte) bool {
+func RangeIncludesHwHash(range_ *pf.RangeSpec, partitionID []byte) bool {
 	var hashed = hwHashPartition(partitionID)
 	return hashed >= range_.KeyBegin && hashed <= range_.KeyEnd
 }
+
+func RangeIncludes(r *pf.RangeSpec, hash uint32) bool {
+	return hash >= r.KeyBegin && hash <= r.KeyEnd
+}
+
+// RangeOverlap is the result of checking whether one Range overlaps another.
+type RangeOverlap int
+
+const (
+	NoRangeOverlap      RangeOverlap = 0
+	PartialRangeOverlap RangeOverlap = 1
+	FullRangeOverlap    RangeOverlap = 2
+)
+
+func RangesOverlap(rangeOne *pf.RangeSpec, rangeTwo *pf.RangeSpec) RangeOverlap {
+	var includesBegin = RangeIncludes(rangeOne, rangeTwo.KeyBegin)
+	var includesEnd = RangeIncludes(rangeOne, rangeTwo.KeyEnd)
+
+	if includesBegin && includesEnd {
+		return FullRangeOverlap
```

Comment:
Just going off the name of this function I think this is incomplete, there should be a "Full Overlap" case if range one is fully contained within range two, but in that case the begin/end keys of range two will not be within range one so the current checks will report "No Overlap".

If the current semantics are intentional the function should probably be something like `RangeContained`.

Reply:
I copied this over from `airbyte` library in `estuary/flow` repository: https://github.com/estuary/flow/blob/master/go/protocols/airbyte/partition_range.go#L131-L144

So I would rather not change the code since there is code depending on this logic as it is. I can rename it though 👍🏽